### PR TITLE
add EventPathMatch to streaming API

### PR DESCRIPTION
**Pull request by camdencheek**
This adds EventPathMatch as a type to our streaming API. This was
previously represented as an EventFileMatch with no line matches. The
intent is to better represent our result domain. Additinally, this
allows us to add metadata to path matches that isn't needed for content
matches.

This just makes the change for streaming events, but I will also
propagate this change down to our internal/search/result types.

Also trying to get these changes in to the streaming API before
we make it available to customers and it needs to be stable

Paired with https://github.com/sourcegraph/src-cli/pull/568

### CODE REVIEW

File: internal/search/streaming/http/events.go [hunk 1]
```diff
@@ -29,6 +29,23 @@ type EventFileMatch struct {
 
 func (e *EventFileMatch) eventMatch() {}
 
+// EventPathMatch is a subset of zoekt.FileMatch for our Event API.
+// It is used for result.FileMatch results with no line matches and
+// no symbol matches, indicating it represents a match of the file itself
+// and not its content.
+type EventPathMatch struct {
```
Comment (camdencheek) [inline internal/search/streaming/http/events.go:36]: This is the new type. We may want to consider putting fields shared between `EventPathMatch`, `EventFileMatch`, and `EventSymbolMatch` into a shared, embedded struct, but for now a little duplication feels okay.

**Review by rvantonder — APPROVED** [review]
```go
var (
   _ Match = (*FileMatch)(nil)
   _ Match = (*RepoMatch)(nil)
   _ Match = (*CommitMatch)(nil)
   )
```

[missing yes?](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/internal/search/result/match.go?L25-29)


So, `File` is a horrible name for what it does. Might this be the right time to make the API cleaner, for ours and customers sake? Not in this PR, but before we release to customers?

- `FileSymbolMatch` -> `SymbolMatch`
- `FileLineMatch` -> `ContentMatch`
- Go side: Instances of `'file'` become `'content'`
- New type: either `FileMatch` or `PathMatch` or `FilePathMatch`

File: internal/search/streaming/http/events.go (hunk 1 above)
Comment (rvantonder) [reply internal/search/streaming/http/events.go:36]: There are only 4 variants, this makes it not worth deduping IMO

**Comment by camdencheek** [comment]
> missing yes? 

Not yet! This doesn't actually create the result type in `internal/search/result`. It just adds the streaming event and converts `result.FileMatch` into one of `http.EventFileMatch`, `http.EventSymbolMatch`, or `http.EventPathMatch`. Next up is propagating that change so our events match our result types. 

> File is a horrible name for what it does

Completely agree. I've been dragging my feet on the `File` -> `Content` change because we can't change the GraphQL names in a backwards-compatible manner, but now that we depend completely on the streaming API for the web app, I agree that it seems like a good time to change everything else to use a more representative name.

**Review by keegancsmith — APPROVED** [review]
LGTM. If I am not mistaken we also need to update the src-cli? cc @stefanhengl

**Comment by stefanhengl** [comment]
Yes, src-cli needs an update, too. It won't break, but it will just ignore the new type.

**Comment by camdencheek** [comment]
Updated src-cli in https://github.com/sourcegraph/src-cli/pull/568

**Pull request merged**